packages/ai-bot/helpers.ts

```diff
@@ -170,6 +176,9 @@ export async function* processStream(stream: AsyncGenerator<string>) {
     // Optimistically start parsing it, and if we hit an error
     // then roll back to the last checkpoint
     if (part == '{') {
+      if (currentMessage) {
+        yield { type: ParsingMode.Break, content: null };
+      }
```

Comment:
For this case you expect currentMessage == ''?

Reply:
I've clarified this in a comment now, and should be clearer with the move away from "Break" as a concept